tests/test_main.py

```diff
@@ -1,6 +1,144 @@
-# TODO: add tests
-from app.main import hello_world
+import pytest
+import ast
+import inspect
+import io
 
+from contextlib import redirect_stdout
 
-def test_hello_world():
-    assert hello_world() == "Hello, world!"
+from app.main import Animal, Herbivore, Carnivore
+
+
+def test_animal_class():
+    assert hasattr(Animal, "alive"), (
+        f"Animal class should have attribute 'alive'"
+    )
+
+
+def test_animal_constructor():
+    lion = Animal("Lion King")
+    assert hasattr(lion, "name"), (
+        "Animal instance should have attribute 'name'"
+    )
+    assert hasattr(lion, "health"), (
+        "Animal instance should have attribute 'health'"
+    )
+    assert hasattr(lion, "hidden"), (
+        "Animal instance should have attribute 'hidden'"
+    )
```

Comment:
Code duplication here - you could add attrs = ("name", "health", "hidden") - for better code quality.
For future